lang/src/accounts/program_owned_account.rs

```diff
@@ -0,0 +1,85 @@
+//! Type validating that the account is owned by the given program
+
+use crate::error::ErrorCode;
+use crate::*;
+use solana_program::account_info::AccountInfo;
+use solana_program::instruction::AccountMeta;
+use solana_program::program_error::ProgramError;
+use solana_program::pubkey::Pubkey;
+use solana_program::system_program;
+use std::collections::BTreeMap;
+use std::ops::Deref;
+
+/// Type validating that the account is owned by the system program
+///
+/// Checks:
+///
+/// - `ProgramOwnedAccount.info.owner == SystemProgram`
+#[derive(Debug, Clone)]
+pub struct ProgramOwnedAccount<'info, T: Id + Clone> {
```

Comment:
does T need to implement `Clone`?

Reply:
Also I expect this to not compile? Would need phantom data?